### PR TITLE
Added option to eject removable device

### LINKED ISSUE #84: Add the option to eject drives
Add an option to eject drives.

### PR DESCRIPTION
**Pull request by tsvietOK**
Fixes #84
Bugs:
- <s>Device cannot be ejected after visiting to it and closing the tab. (help needed)</s>
- <s>Missing menu icon</s>
- <s>Missing translation(will be added after fixing the first issue)</s>

Previous PR with the same issue: #1810

### CODE REVIEW

**Comment by gave92** [comment]
"Device cannot be ejected after visiting to it and closing the tab. (help needed)"
We are not properly calling CloseHandle in this function:
ItemViewModel -> `CheckFolderForHiddenAttribute`
Last time we had this issue, closing all pending handles solved it

**Comment by XPoppyX** [comment]
Icon proposal, i moved it before "Properties"

Captura de pantalla 2020-11-13 180738

**Comment by tsvietOK** [comment]
@XPoppyX What code of this icon?

**Comment by duke7553** [comment]
@gave92 You're absolutely right about the redundant calls to FindFirstFile. We certainly don't need to use that parent method to find the hidden attributes there. That was my mistake 😳

**Comment by XPoppyX** [comment]
@tsvietOK  It is a custom icon that I added now, if you like the icon I can add a PR in your repository, Also, if you agree, with the change about moving it above properties

**Comment by tsvietOK** [comment]
@XPoppyX Agreed with both changes.

**Comment by XPoppyX** [comment]
@tsvietOK Done

**Comment by yaichenbaum** [comment]
@tsvietOK Can you make sure we exlude the eject option in process.cs that is detected from File Explorer?

**Comment by tsvietOK** [comment]
@yair100 There is no eject option in File Explorer(in regular folders). there only two way to eject device: through tray and in My computer.  And there no file process.cs.

File: Files/Interacts/RemovableDevice.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+using System.Diagnostics;
+using System.Threading.Tasks;
+using static Files.Helpers.NativeIoDeviceControlHelper;
+
+namespace Files.Interacts
+{
+    public class RemovableDevice
+    {
+        private IntPtr handle;
+        private string driveLetter;
```
Comment (d2dyno1) [inline Files/Interacts/RemovableDevice.cs:11]: Why not `private char driveLetter;`?

**Comment by yaichenbaum** [comment]
@tsvietOK It's program.cs, you can find it over here https://github.com/gave92/files-uwp/blob/master/Files.Launcher/Program.cs#L369.

**Comment by tsvietOK** [comment]
@yair100 Sorry, i don't understand what are you talking about. Where is connection between Program.cs and Sidebar context menu, which is not taken from Explorer context menu?

**Comment by yaichenbaum** [comment]
@tsvietOK Add "Eject" to the known items list, otherwise the context menu will have two eject options.

**Comment by tsvietOK** [comment]
@yair100 There is only one eject option.
image

**Comment by yaichenbaum** [comment]
@tsvietOK There is another one if you right click inside the root path of a drive.

**Comment by tsvietOK** [comment]
@yair100 In context menu of the drive such option exists, but it is not duplicated. There is two ways to eject device.
image

**Review by yair100 — APPROVED** [review]

**Pull request merged**